Core/Src/main.c

```diff
@@ -87,17 +90,44 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
+
+  // Starts PWM
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  uint8_t DataReceived[DATASIZE] = {0};
+  uint8_t DataTransmitted[DATASIZE] = {0};
+  uint16_t bit_mask = 0b0000001111111111;
+
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  // Sets CS line to low
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+	  // Uses appropriate SPI protocol to transmit and receive ADC values
+	  HAL_SPI_TransmitReceive(&hspi1, DataTransmitted, DataReceived, sizeof(DataTransmitted), 24);
     /* USER CODE END WHILE */
 
+	  // Delay between processes to ensure that MCU doesnt overload
+	  HAL_Delay(10);
+
     /* USER CODE BEGIN 3 */
+
+	  // Taking the 10 LSB's (B1 to B9) to read from
+	  uint16_t adc_value = DataReceived[1] & bit_mask;
+
+	  // Converting ADC value to number of counts
+	  uint16_t On_Counts = (adc_value * (2^10)) / 3;
```

Comment:
don't think you can do exponential operation with `^`. Since I can't check the timer configuration in the `tim.c` file, I couldn't confirm if you have the right on counts here. Please also push the timer files later. 
The formula here is also a bit off. The BootCamp requires an output pwm duty cycle that varies from 5-10%, and 5% maps to the `adc_value` to be 0, 10% maps to the `adc_value` to be 2^10-1. Your formula should look like `on_counts of 5% duty cycle + (adc_value/max_adc_value * on_counts of 5% duty cycle )` if this makes sense.